src/main/java/com/avispl/symphony/dal/avdevices/camera/panasonic/awue150/CameraPanasonicAWUE150Communicator.java

```diff
@@ -0,0 +1,1779 @@
+				getDefaultValueForNullData(String.valueOf(cachedLiveCameraInfo.getZoomUIValue()), DeviceConstant.NONE));
+		stats.put(groupName.concat(ZoomControlMetric.ZOOM_CONTROL_SPEED.getName()).concat(ZoomControlMetric.CURRENT_VALUE.getName()), String.valueOf((int) cachedZoomControlSpeed));
+	}
+
+	/**
+	 * This method is used to handle zoom control
+	 *
+	 * @param stats store all statistics
+	 * @param advancedControllableProperties store all controllable properties
+	 * @param controllableProperty controllable property
+	 * @param value value of controllable property
+	 */
+	private void zoomControl(Map<String, String> stats, List<AdvancedControllableProperty> advancedControllableProperties, String controllableProperty, String value) throws FailedLoginException {
+		ZoomControlMetric zoomControlMetric = ZoomControlMetric.getByName(controllableProperty);
+		isEmergencyDelivery = true;
+		switch (zoomControlMetric) {
+			case ZOOM_CONTROL:
+				String currentValueInHex = convertFromUIValueToAPIValue(Float.parseFloat(value), DeviceConstant.MAX_ZOOM_API_VALUE, DeviceConstant.MIN_ZOOM_API_VALUE, DeviceConstant.MAX_ZOOM_UI_VALUE,
+						DeviceConstant.MIN_ZOOM_UI_VALUE);
+
+				String command = Command.HASH.concat(Command.ZOOM.concat(currentValueInHex.toUpperCase()));
+				String responseValueInHex = performCameraControl(command, controllableProperty, value, DeviceURL.CAMERA_PTZ_CONTROL, true);
+				cachedLiveCameraInfo.setZoomPosition(responseValueInHex.substring(Command.ZOOM.length()));
+
+				// camera need 3s to apply new value
+				try {
+					Thread.sleep(3000);
+				} catch (InterruptedException e) {
+					e.printStackTrace();
+				}
+				retrieveSimultaneous();
+				break;
+			case ZOOM_CONTROL_FAR:
+				Float currentUIValue = convertFromApiValueToUIValue(cachedLiveCameraInfo.getZoomPosition(),
+						DeviceConstant.MAX_ZOOM_API_VALUE, DeviceConstant.MIN_ZOOM_API_VALUE, DeviceConstant.ZOOM_UI_API_CONVERT_FACTOR, DeviceConstant.MIN_ZOOM_UI_VALUE) + cachedZoomControlSpeed;
+				if (currentUIValue > DeviceConstant.MAX_ZOOM_UI_VALUE) {
+					currentUIValue = DeviceConstant.MAX_ZOOM_UI_VALUE;
+				}
+				if (currentUIValue < DeviceConstant.MIN_ZOOM_UI_VALUE) {
+					currentUIValue = DeviceConstant.MIN_ZOOM_UI_VALUE;
+				}
+
+				currentValueInHex = convertFromUIValueToAPIValue(currentUIValue, DeviceConstant.MAX_ZOOM_API_VALUE, DeviceConstant.MIN_ZOOM_API_VALUE, DeviceConstant.MAX_ZOOM_UI_VALUE,
+						DeviceConstant.MIN_ZOOM_UI_VALUE);
+				command = Command.HASH.concat(Command.ZOOM.concat(currentValueInHex.toUpperCase()));
+				responseValueInHex = performCameraControl(command, controllableProperty, value, DeviceURL.CAMERA_PTZ_CONTROL, true);
+				cachedLiveCameraInfo.setZoomPosition(responseValueInHex.substring(Command.ZOOM.length()));
+
+				// camera need 3s to apply new value
+				try {
+					Thread.sleep(3000);
+				} catch (InterruptedException e) {
+					e.printStackTrace();
```

Comment:
exception processing here, similar to above

Reply:
I fixed it